### PR TITLE
Python 3 compatibility + Scouting updates

### PR DESCRIPTION
**Pull request by a1ph4g33k**
Lots of clean up.

Formatted source with black.
Fixed all linter & pydocstyle errors.
All tests pass.
Updated requirements.txt to current revisions of things.
All work done using Python 3.7.4
All Python2 work items resolved:
- use of xrange
- bytes-like object handling
- string.encode('base64')
- explicit checking for None

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -5,51 +5,51 @@
...
-    ],
-    scripts = [],
+setup(
+    name="bloomfilter",
+    version="0.2.0",
+    description="Bloom Filter (Python)",
+    license="MIT License",
+    url="http://github.com/seomoz/bloomfilter-py",
+    author="Big Data",
+    author_email="vadim@moz.com",
```
Comment (phil-hildebrand) [inline setup.py]: Should we consider changing this to a group email?  (perhaps whatever the current pro services team email is?)

**Comment by a1ph4g33k** [comment]
Many repositories these days are making use of a Python formatter called black. I am a supporter of it and the formatting changes it makes.  

https://black.readthedocs.io/en/stable/the_black_code_style.html#strings  If that helps.

File: setup.py (hunk 1 above)
Comment (a1ph4g33k) [reply setup.py]: ... that *is* a bad email, what should we replace it with ?

Comment (a1ph4g33k) [reply setup.py]: Checking with @seomozSLE

Comment (a1ph4g33k) [reply setup.py]: Updated.

**Pull request merged**